### PR TITLE
initialize errors HashSet without high initial capacity

### LINKED ISSUE #447: Hash Table Overflow
**Describe the bug**
Rustscan crashes with a hash table overflow

**To Reproduce**
Run: rustscan -a 10.0.0.0/8 --top

**Expected behavior**
It works

**Console Output**
```
Automatically increasing ulimit value to 1000.
thread 'main' panicked at 'Hash table capacity overflow', /cargo/registry/src/github.com-1ecc6299db9ec823/hashbrown-0.12.0/src/raw/mod.rs:96:40
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
zsh: abort      rustscan -a 10.0.0.0/8 --top
```
**Desktop (please complete the following information):**
 - OS: Kali Linux ARM64
 - Version rustscan 2.1.0
 - Device: Raspberry Pi 4

**Additional context**
Compiled for Arm64 from latest release

### LINKED ISSUE #523: Rustscan crash with failed memory allocation
**Describe the bug**
Rustscan crash with failed memory allocation

**To Reproduce**
```
rustscan -a 10.0.0.0/10 --ulimit 5000
```

Note that with lower mask value it doesn't crash (limit for me is `/13`)

**Expected behavior**
Maybe display a cleaner error like "Cannot scan this many addresses."

**Screenshots**
image

**Desktop (please complete the following information):**
- rustscan 2.1.1 (from nix)

### PR DESCRIPTION
**Pull request by fabianbees**
In most cases it is not expected to have that many errors, which causes an increase of the HashSet, it shouldn't be necessary to pre-allocate that much memory for potential errors.

fixes #447
fixes #523

### CODE REVIEW

**Review by bee-san — CHANGES_REQUESTED** [review]
commenting to see if CI will run... it is not?

**Comment by bee-san** [comment]
Hi! Mind pushing an empty commit? CI has failed to pick this up and I think it'll only pick it up on a new commit? 🤔

**Comment by fabianbees** [comment]
CI is not running, because the Build-Workflow somehow requires an approval from a maintainer:
image

**Comment by bee-san** [comment]
> CI is not running, because the Build-Workflow somehow requires an approval from a maintainer: image

Yes, that's fine. Before it wasn't running at all, and the emtpy commit kicked GitHub into seeing it had _something_ to do here :)

CI is running!

**Comment by bee-san** [comment]
thank you!!

**Comment by fabianbees** [comment]
Your welcome! :)

**Pull request merged**
